go-controller/pkg/node/default_node_network_controller_test.go

```diff
@@ -652,4 +662,365 @@ var _ = Describe("Node", func() {
 			Expect(err).NotTo(HaveOccurred())
 		})
 	})
+	Describe("node route management", func() {
+		var (
+			testNS ns.NetNS
+			nc     *DefaultNodeNetworkController
+			app    *cli.App
+		)
+
+		const (
+			nodeName       = "my-node"
+			remoteNodeName = "other-node"
+		)
+
+		BeforeEach(func() {
+			var err error
+			testNS, err = testutils.NewNS()
+			Expect(err).NotTo(HaveOccurred())
+			Expect(config.PrepareTestConfig()).To(Succeed())
+
+			app = cli.NewApp()
+			app.Name = "test"
+			app.Flags = config.Flags
+		})
+
+		AfterEach(func() {
+			util.ResetNetLinkOpMockInst() // other tests in this package rely directly on netlink (e.g. gateway_init_linux_test.go)
+			Expect(testNS.Close()).To(Succeed())
+		})
+
+		Context("with a cluster in IPv4 mode", func() {
+			const (
+				ethName     string = "eth0"
+				nodeIP      string = "169.254.254.60"
+				ethCIDR     string = nodeIP + "/24"
+				otherNodeIP string = "169.254.254.61"
+				fullMask           = 32
+			)
+			// var eth0MAC string
+			var link netlink.Link
+
+			BeforeEach(func() {
+				config.IPv4Mode = true
+				config.IPv6Mode = false
+
+				// Note we must do this in default netNS because
+				// nc.WatchNodes() will spawn goroutines which we cannot lock to the testNS
```

Comment:
hmm.. youre creating a link on the test network namespace

Reply:
yeah oops, i should make it some dummy loopback interface...will change it